### PR TITLE
[SPARK-42598][TEST] Refactor TPCH schema to separate file similar to TPCDS for code reuse

### PR DESCRIPTION
**Pull request by Surbhi-Vijay**
### What changes were proposed in this pull request?
Changes are only in tests files.
- Refactored the `TPCHBase` class and created `TPCHSchema` similar to TPCDS.
- Created a base class `TPCSchema` which is extended by `TPCDSSchema` and `TPCHSchema`.

### Why are the changes needed?
This PR just refactors the code. Going forward it will help in making code changes at just one place to reflect in both TPCDS and TPCH. 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Ran the PlanStabilitySuite to verify the changes.

### CODE REVIEW

**Comment by Surbhi-Vijay** [comment]
Can someone please help in creating issue for this pull request ? I do not have a ASF Jira account.

**Comment by bjornjorgensen** [comment]
[private@spark.apache.org](mailto:private@spark.apache.org)

**Comment by Surbhi-Vijay** [comment]
retest this please

**Comment by Surbhi-Vijay** [comment]
Gentle ping!! @gengliangwang  @cloud-fan @viirya @Ngone51 @maropu @HyukjinKwon.

**Comment by bjornjorgensen** [comment]
you must enable github action on yours repo https://github.com/apache/spark/pull/40171/checks?check_run_id=11594767505

File: sql/core/src/test/scala/org/apache/spark/sql/TPCSchema.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package org.apache.spark.sql
+
+/**
+ * Base trait for TPC related tests.
+ *
+ * https://www.tpc.org/tpc_documents_current_versions/current_specifications5.asp
+ */
+trait TPCSchema {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/TPCSchema.scala:25]: this trait doesn't seem to provide any value. Is there a place that matches `TPCSchema` and call its methods?

Comment (Surbhi-Vijay) [reply sql/core/src/test/scala/org/apache/spark/sql/TPCSchema.scala:25]: The fields in this trait were required in both TPCHSchema as well as in TPCDSSchema. So, created this new trait to keep the info at single place. Both TPCHSchema and TPCDSSchema extends from this trait.

**Pull request closed without merging**